src/test/ripple-libpp_demo.cpp

```diff
@@ -0,0 +1,151 @@
+    return std::make_shared<STTx const>(std::ref(sitTrans));
+}
+
+bool demonstrateSigning(ripple::KeyType keyType, std::string seedStr,
+    std::string expectedAccount)
+{
+    using namespace ripple;
+
+    auto const seed = parseGenericSeed(seedStr);
+    assert(seed);
+    auto const keypair = generateKeyPair(keyType, *seed);
+    auto const id = calcAccountID(keypair.first);
+    assert(toBase58(id) == expectedAccount);
+
+    STTx noopTx(ttACCOUNT_SET,
+        [&](auto& obj)
+    {
+        obj[sfAccount] = id;
+        obj[sfFee] = STAmount{ 100 };
+        obj[sfFlags] = tfFullyCanonicalSig;
+        obj[sfSequence] = 18;
+        obj[sfSigningPubKey] = keypair.first.slice();
+    });
+
+    std::cout << "\nBefore signing: \n" << noopTx << "\n"
+        << "Serialized: " << noopTx.getJson(0, true)[jss::tx] << "\n";
+
+    noopTx.sign(keypair.first, keypair.second);
+
+    auto const serialized = serialize(noopTx);
+    std::cout << "\nAfter signing: \n" << noopTx << "\n"
+        << "Serialized: " << serialized << "\n";
+
+    auto const deserialized = deserialize(serialized);
+    assert(deserialized);
+    assert(deserialized->getTransactionID() == noopTx.getTransactionID());
+    std::cout << "Deserialized: " << *deserialized << "\n";
+
+    auto const check1 = noopTx.checkSign(false);
+
+    std::cout << "Check 1: " << (check1.first ? "Good" : "Bad!") << "\n";
+    assert(check1.first);
+
+    // Use the function primitives, which are hidden by the `STTx`
+    // interface, to check the signature again and verify that
+    // `STTx` is working properly.
+    auto const& signatureSlice = noopTx[sfTxnSignature];
+    Blob const data = [&]
+    {
+        // This is a copy of the static `getSigningData` function body
+        // which is needed by `verify`.
+        Serializer s;
+        s.add32(HashPrefix::txSign);
+        noopTx.addWithoutSigningFields(s);
+        return s.getData();
+    }();
+
+    // STTx::checkSign calls `verify` indirectly via `checkSingleSign`
+    auto const check2 = verify(
+        keypair.first,
+        makeSlice(data),
+        signatureSlice,
+        true);
+
+    std::cout << "Check 2: " << (check2 ? "Good" : "Bad!") << "\n";
+
+    return check1.first && check2;
+}
+
+int main (int argc, char** argv)
+{
+#if defined(__GNUC__) && !defined(__clang__)
+    auto constexpr gccver = (__GNUC__ * 100 * 100) +
+                            (__GNUC_MINOR__ * 100) +
+                            __GNUC_PATCHLEVEL__;
+
+    static_assert (gccver >= 50100,
+        "GCC version 5.1.0 or later is required to compile rippled.");
+#endif
+
+    static_assert (BOOST_VERSION >= 105700,
+        "Boost version 1.57 or later is required to compile rippled");
+
+    auto const pass1 = demonstrateSigning(ripple::KeyType::secp256k1,
+        "alice", "rG1QQv2nh2gr7RCZ1P8YYcBUKCCN633jCn");
+
+    auto const pass2 = demonstrateSigning(ripple::KeyType::ed25519,
+        "alice", "r9mC1zjD9u5SJXw56pdPhxoDSHaiNcisET");
```

Comment:
Maybe this is out of scope for this project, but:

1. Should we accept JSON or a hexblob from the command line so users can specify ready-made transactions?
2. Should we demonstrate more transactions than `ttACCOUNT_SET` which is rare? Considering the esoteric syntax for creating a transaction (constructor which accepts a lambda callback that's invoked to populate the fields) it might be a good idea.